Update utils.py

AeosHere opened the pull request.

AeosHere committed:
Update utils.py

shishu314 commented on utils.py:
```diff
@@ -20,8 +20,57 @@ def ValidateUser(username, password):
+    connection = DatabaseConnect()
+    try:
+        with connection.cursor() as cursor:
+            sql = "CALL GetCoursesByUser(" + id + ");"
```
Why do you make a CALL here and then do cursor.callproc later?

shishu314 commented on utils.py:
```diff
@@ -20,8 +20,57 @@ def ValidateUser(username, password):
+    connection = DatabaseConnect()
+    try:
+        with connection.cursor() as cursor:
+            sql = "CALL GetCoursesByUsername('" + id + "');"
```
same here, it should be consistent

shishu314 commented on utils.py:
```diff
@@ -20,8 +20,57 @@ def ValidateUser(username, password):
+        connection.commit()
+        connection.close()
+
 #Tests for this method
```
If you're going to clump all the test statements together you might as well delete my comment "Tests for this method". Also just use triple quote instead of # signs

shishu314 reviewed (changes requested):
look over my comments, coding style inconsistency, functionality looks good.

AeosHere committed:
Update utils.py

shishu314 commented on utils.py:
```diff
@@ -20,8 +20,55 @@ def ValidateUser(username, password):
+    try:
+        with connection.cursor() as cursor:
+            cursor.callproc("RegisterStudent", [UName, PW, FName, "@Success"])
+            success = cursor.execute("SELECT @Success")
```
You have to do a .fetchone()

shishu314 commented on utils.py:
```diff
@@ -20,8 +20,55 @@ def ValidateUser(username, password):
+    try:
+        with connection.cursor() as cursor:
+            cursor.callproc("AddRating", [UName, CourseID, Sem, Section, Rate, Note, "@Success"])
+            success = cursor.execute("SELECT @Success")
```
same here

AeosHere committed:
Update utils.py

#FIXUP
For RegisterStudent, try to use #cursor.callproc("RegisterStudent", [UName, PW, FName, "@Success"]) in the future
For AddRating, try to use #cursor.callproc("AddRating", [UName, CourseID, Sem, Section, Rate, Note, "@Success"]) in the future

AeosHere commented:
#FIXUP
For RegisterStudent, try to use #cursor.callproc("RegisterStudent", [UName, PW, FName, "@Success"]) in the future
For AddRating, try to use #cursor.callproc("AddRating", [UName, CourseID, Sem, Section, Rate, Note, "@Success"]) in the future

AeosHere committed:
Update utils.py

AeosHere committed:
Update utils.py
